site-builder/src/publish.rs

```diff
@@ -26,22 +30,34 @@ use crate::{
     Config,
 };
 
-pub async fn publish_site(
-    directory: &Path,
-    content_encoding: &ContentEncoding,
-    site_name: &str,
-    config: &Config,
-    epochs: u64,
-    preprocess: bool,
-) -> Result<()> {
+#[derive(Parser, Debug)]
+pub struct PublishOptions {
+    /// The directory containing the site sources.
+    pub directory: PathBuf,
+    #[clap(short = 'e', long, value_enum, default_value_t = ContentEncoding::PlainText)]
+    /// The encoding for the contents of the site's resources.
+    pub content_encoding: ContentEncoding,
+    /// The name of the site.
+    #[clap(short, long, default_value = "test site")]
+    pub site_name: String,
```

Comment:
This is currently not used in the update function. I'd suggest either moving this out from the `PublishOptions` struct and to the options of the `Publish` command, or alternatively support changing the site name in an update.